types/react-dom/experimental.d.ts

```diff
@@ -50,4 +52,25 @@ declare module '.' {
     type FormStatus = FormStatusPending | FormStatusNotPending;
 
     function experimental_useFormStatus(): FormStatus;
+
+    function experimental_useFormState<State>(
+        action: (state: State) => Promise<State>,
+        initialState: State,
+        permalink?: string,
+    ): [state: State, dispatch: () => void];
+    function experimental_useFormState<State, Payload>(
+        action: (state: State, payload: Payload) => Promise<State>,
```

Comment:
We changed our mind and we're going to allow a sync action to be passed here, too.

The updated type would be `(state: State, payload: Payload) => State | Promise<State>`.